core/network/impl/router_libp2p.cpp

```diff
@@ -144,22 +156,49 @@ namespace kagome::network {
         babe_observer_->onBlockAnnounce(msg_res.value());
         return true;
       }
-      case MsgType::CONSENSUS: {
-        auto vote_msg_res =
-            scale::decode<consensus::grandpa::VoteMessage>(msg.data);
-        if (vote_msg_res) {
-          grandpa_observer_->onVoteMessage(vote_msg_res.value());
-          return true;
-        }
+      case GossipMessage::Type::CONSENSUS: {
+        auto grandpa_msg_res = scale::decode<GrandpaMessage>(msg.data);
 
-        auto fin_msg_res = scale::decode<consensus::grandpa::Fin>(msg.data);
-        if (fin_msg_res) {
-          grandpa_observer_->onFinalize(fin_msg_res.value());
-          return true;
+        if (not grandpa_msg_res) {
+          log_->error("error while decoding a consensus (grandpa) message: {}",
+                      grandpa_msg_res.error().message());
+          return false;
         }
 
-        log_->error("error while decoding a consensus message");
-        return false;
+        auto &grandpa_msg = grandpa_msg_res.value();
+
+        return visit_in_place(
+            grandpa_msg,
+            [this, &peer_id](const network::GrandpaVoteMessage &vote_message) {
+              grandpa_observer_->onVoteMessage(peer_id, vote_message);
+              return true;
+            },
+            [this, &peer_id](const network::GrandpaPreCommit &fin_message) {
+              grandpa_observer_->onFinalize(peer_id, fin_message);
+              return true;
+            },
+            [this, &peer_id](const GrandpaNeighborPacket &neighbor_packet) {
+              BOOST_ASSERT_MSG(false,
+                               "Unimplemented variant (GrandpaNeighborPacket) "
+                               "of consensus (grandpa) message");
+              (void)this;
+              (void)peer_id;
```

Comment:
Why we are doing it here?

Reply:
It's way supress unused-variable warning. Will be removed out of lambda's capture list